doc/api/net.md

```diff
@@ -578,259 +753,269 @@ the socket is destroyed (for example, if the client disconnected).
+* `options` {Object}
+* `connectListener` {Function}
+
+The `net.createConnection()` method creates a new [`net.Socket`][],
+automatically opens the socket, then returns it.
 
-The options are passed to both the [`net.Socket`][] constructor and the
-[`socket.connect`][] method.
+The `options` are passed to both the [`net.Socket`][] constructor and the
+[`socket.connect`][] method. The options relevant to each may be used.
 
-The `connectListener` parameter will be added as a listener for the
-[`'connect'`][] event once.
+The `connectListener` parameter will be added as a one-time listener for the
+[`'connect'`][] event.
 
-Here is an example of a client of the previously described echo server:
+The following example illustrates a client for an echo server:
 
 ```js
 const net = require('net');
+
 const client = net.createConnection({port: 8124}, () => {
   //'connect' listener
   console.log('connected to server!');
   client.write('world!\r\n');
-});
-client.on('data', (data) => {
+}).on('data', (data) => {
   console.log(data.toString());
   client.end();
-});
-client.on('end', () => {
+}).on('end', () => {
   console.log('disconnected from server');
 });
 ```
 
-To connect on the socket `/tmp/echo.sock` the second line would just be
-changed to
-
-```js
-const client = net.connect({path: '/tmp/echo.sock'});
-```
-
 ## net.createConnection(path[, connectListener])
 <!-- YAML
 added: v0.1.90
 -->
 
-A factory function, which returns a new unix [`net.Socket`][] and automatically
-connects to the supplied `path`.
+* `path` {String}
+* `connectListener` {Function}
+
+The `net.createConnection()` method creates a new [`net.Socket`][],
+automatically opens the socket using `path`, then returns it.
 
-The `connectListener` parameter will be added as a listener for the
-[`'connect'`][] event once.
+The `connectListener` parameter will be added as a one-time listener for the
+[`'connect'`][] event.
 
 ## net.createConnection(port[, host][, connectListener])
 <!-- YAML
 added: v0.1.90
 -->
 
-A factory function, which returns a new [`net.Socket`][] and automatically
-connects to the supplied `port` and `host`.
+* `port` {number}
+* `host` {String}. Defaults to `localhost`
+* `connectListener` {Function}
 
-If `host` is omitted, `'localhost'` will be assumed.
+The `net.createConnection()` method creates a new [`net.Socket`][],
+automatically opens the socket for the given `port` and `host`, then returns it.
 
-The `connectListener` parameter will be added as a listener for the
-[`'connect'`][] event once.
+The `connectListener` parameter will be added as a one-time listener for the
+[`'connect'`][] event.
 
 ## net.createServer([options][, connectionListener])
 <!-- YAML
 added: v0.5.0
 -->
 
-Creates a new server. The `connectionListener` argument is
-automatically set as a listener for the [`'connection'`][] event.
+* `options` {Object}
+  * `allowHalfOpen` {boolean} If `true`, the socket will not automatically send
```

Comment:
It might be a good idea to do something like `the socket associated with each incoming connection` here, like it’s done for `pauseOnConnect`